// amadeus/lib/amadeus/namespaces/reference_data.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _urls = _interopRequireDefault(require("./reference_data/urls"));

var _locations = _interopRequireDefault(require("./reference_data/locations"));

var _location = _interopRequireDefault(require("./reference_data/location"));

var _airlines = _interopRequireDefault(require("./reference_data/airlines"));

var _recommended_locations = _interopRequireDefault(require("./reference_data/recommended_locations"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * A namespaced client for the
 * `/v2/reference-data` endpoints
 *
 * Access via the {Amadeus} object
 *
 * ```js
 * let amadeus = new Amadeus();
 * amadeus.referenceData.urls;
 * ```
 *
 * @param {Client} client
 * @property {Urls} urls
 * @protected
 */
var ReferenceData =
/*#__PURE__*/
function () {
  function ReferenceData(client) {
    _classCallCheck(this, ReferenceData);

    this.client = client;
    this.urls = new _urls["default"](client);
    this.locations = new _locations["default"](client);
    this.airlines = new _airlines["default"](client);
    this.recommendedLocations = new _recommended_locations["default"](client);
  }
  /**
   * The namespace for the Location APIs - accessing a specific location
   *
   * @param  {string} [locationId]  The ID of the location to search for
   * @return {Location}
   **/


  _createClass(ReferenceData, [{
    key: "location",
    value: function location(locationId) {
      return new _location["default"](this.client, locationId);
    }
  }]);

  return ReferenceData;
}();

var _default = ReferenceData;
exports["default"] = _default;
module.exports = exports["default"];
